### DISCUSSION TITLE
Using google sheets API

### DISCUSSION THREAD

**Discussion by Arshkamran1** [deta/cloud-docs#722] [Help]
Why doesn't deta install the google api's. I created a program in python that converts a google sheet to a data frame, and then emails the last added row. However although it works locally, on the server I get these issues. Traceback (most recent call last):
  File "/opt/python/detalib/debugger.py", line 142, in wrap
    result = func(event, context)
  File "/var/task/_entry.py", line 12, in handler
    import main  # noqa
  File "/var/task/main.py", line 1, in <module>
    from googleapiclient.discovery import build
ModuleNotFoundError: No module named 'googleapiclient', Someone plz help me

**Comment by sudo-subg** [comment]
As per the docs [here](https://docs.deta.sh/docs/micros/about/#important-notes):
> For unknown reasons, Google and Firebase packages for Python do not install successfully on Micros.

But when tried to install [Google API Client](https://pypi.org/project/google-api-python-client/) it installed successfully.
![image](https://user-images.githubusercontent.com/102468247/212261885-8d4ae691-4ab0-4aad-9b40-fdfab13d7dbc.png)

**Reply by Arshkamran1** [reply]
How did u install it, may I ask did you just put it in the requirements txt?

**Reply by Arshkamran1** [reply]
If so what did you put, and how do you check the dependencies

**Reply by sudo-subg** [reply]
> How did u install it, may I ask did you just put it in the requirements txt?

You add the required dependencies in requirements.txt as you should and in this case `google-api-python-client`.

> If so what did you put, and how do you check the dependencies

To check the dependencies of the micro you can run `deta details` in the micro's folder's root directory or check in your micros details using [Deta Dashboard](https://web.deta.sh).

**Reply by Arshkamran1** [reply]
pandas
python-dotenv
google-api-python-client 

this is what is inside my requirments.txt and I did deta deploy but still can't check what dependencies I have
when I run deta details I get this 

{
        "name": "send_emailauto",
        "id": "7ec065aa-da78-48c5-86ee-eb1c939174ff",
        "project": "default",
        "runtime": "python3.9",
        "endpoint": "https://1ed4gb.deta.dev",
        "region": "us-east-1",
        "environment_variables": [
                "EMAIL",
                "PASSWORD",
                "APP_PASSWORD"
        ],
        "visor": "enabled",
        "http_auth": "disabled",
        "cron": "0/5 * * * ? *"

**Reply by Arshkamran1** [reply]
nvm I had spelled requirements wrong lol thnx for your help
